pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -193,15 +193,20 @@ func (service *SecurityPolicyService) CreateOrUpdateSecurityPolicy(obj *v1alpha1
 		}
 
 		if (finalProjectShares != nil) && len(*finalProjectShares) != 0 {
-			err = service.shareStore.Apply(finalProjectShares)
+			if err = service.shareStore.Apply(finalProjectShares); err != nil {
+				return err
```

Comment:
add a log.Error(err, "failed to apply store", "project", finalProjectShares) ?

Reply:
Added.